client/driver/logs.go

```diff
@@ -0,0 +1,146 @@
+package driver
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+const (
+	bufSize = 32 * 1024 // Max number of bytes read from a buffer
+)
+
+// LogRotator rotates files for a buffer and retains only the last N rotated
+// files
+type LogRotator struct {
+	maxFiles   int
+	fileSize   int64
+	path       string
+	fileName   string
+	logFileIdx int
+
+	logger *log.Logger
+}
+
+// NewLogRotator configures and returns a new LogRotator
+func NewLogRotator(path string, fileName string, maxFiles int, fileSize int64, logger *log.Logger) (*LogRotator, error) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	logFileIdx := 0
+	for _, f := range files {
+		if strings.HasPrefix(f.Name(), fileName) {
+			fileIdx := strings.TrimPrefix(f.Name(), fmt.Sprintf("%s.", fileName))
+			n, err := strconv.Atoi(fileIdx)
+			if err != nil {
+				continue
+			}
+			if n > logFileIdx {
+				logFileIdx = n
+			}
+		}
+	}
+
+	return &LogRotator{
+		maxFiles:   maxFiles,
+		fileSize:   fileSize,
+		path:       path,
+		fileName:   fileName,
+		logFileIdx: logFileIdx,
+		logger:     logger,
+	}, nil
+}
+
+// Start reads from a Reader and writes them to files and rotates them when the
+// size of the file becomes equal to the max size configured
+func (l *LogRotator) Start(r io.Reader) error {
+	buf := make([]byte, bufSize)
+	for {
+		logFileName := filepath.Join(l.path, fmt.Sprintf("%s.%d", l.fileName, l.logFileIdx))
+		remainingSize := l.fileSize
```

Comment:
Put comments above each block of code describing what it does.

* Determine space left on this file
* Write logs to file until it is full